### PR TITLE
[Bug] Fix condition for disabling controls in product cards

### LINKED ISSUE #10216: [Order edition] Product card editable controls not disabled when order is not editable
**Describe the bug**
When an order is marked as complete, sections like products, custom amounts, and payments are locked and become non-editable. However, despite the product section showing a lock icon, it remains editable – allowing for changes in quantity, product removal, or additions.

**Screenshots**

https://github.com/woocommerce/woocommerce-android/assets/4527432/461202ef-891b-4e23-a7cc-4bf671c54185

**Expected behavior**
We should not be able to remove the product or modify its quantity.

### PR DESCRIPTION
**Pull request by samiuelson**
Closes: #10216 

### Description

This PR disables editable controls in order's product cards in case the order is not editable (e.g. it's in `completed` state, and the lock icon is visible).

### Testing instructions

1. Create order and set it's state to `completed`
2. Open the order and click `Edit`
3. Verify you it's impossible to edit products

### Images/gif

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -6,6 +6,7 @@
 -----
 - [**] Introduce ability to start creating a product by sharing images from other apps [https://github.com/woocommerce/woocommerce-android/pull/10331]
 - [*] [Internal] Improvements to the performance and internal logic of the barcode scanning screen [https://github.com/woocommerce/woocommerce-android/pull/10307]
+- [*] [Internal] Fixed disabling editable controls in product cards in order edit screen in case order is not editable [https://github.com/woocommerce/woocommerce-android/pull/10345]
```
Comment (kidinov) [inline RELEASE-NOTES.txt]: That's 16.6, but the PR goes to 16.7 according to the milestone

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt [hunk 2]
```diff
@@ -620,7 +620,8 @@ private fun AmountPicker(
                 modifier = Modifier
                     .padding(horizontal = dimensionResource(id = R.dimen.minor_25))
                     .widthIn(min = 12.dp, max = 128.dp)
-                    .width(IntrinsicSize.Min)
+                    .width(IntrinsicSize.Min),
+                enabled = isAmountChangeable,
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt:617]: Just trying to understand the goal here. We already have `readOnly = !isAmountChangeable,` which is kinda the same but the field is still focusable, right? I am wondering if you want to make it completely disabled, do we need `readOnly` property to be set then?

**Review by kidinov — APPROVED** [review]
LGTM. I left one kinda blocking comment regarding the release notes, and one question

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt (hunk 2 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt:617]: Without `enabled` there is the `textStyle` animation is enabled when tapping the text field

https://github.com/woocommerce/woocommerce-android/assets/4527432/12a81a51-831d-4136-a504-844faba2d095

I replaced `readOnly` with `enabled` – d9c187a8bf986c0992197806c5c90a40abe1e80e

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (samiuelson) [reply RELEASE-NOTES.txt]: Fixed – d83bcd86b77ae068dee1c908e4280d1c05f6a63d

**Comment by samiuelson** [comment]
> LGTM. I left one kinda blocking comment regarding the release notes, and one question

Thanks for review @kidinov! I've addressed your comments 👍🏼

**Pull request merged**